### ISSUE TITLE
continuous training using finetuner

### ISSUE THREAD

**Issue by bwanglzu** [jina-ai/finetuner#627]
Issue reported from @prafull 

After we dump the fine-tuned model with torch. We should also save the `optimizer` , `loss` etc, and allow user to continuous training with new data.

**Comment by guenthermi** [comment]
We discussed, that we only want to focus on continue training by loading the model weight but newly initializing the optimizer. To store checkpoints online to retain them for later after the fine-tuning job has terminated, we might want to use the TrainingCheckpoint callback, but this is out-of-scope for now.
